addons/coupon/wizard/coupon_generate.py

```diff
@@ -17,6 +17,7 @@ class CouponGenerate(models.TransientModel):
         ], default='nbr_coupon')
     partners_domain = fields.Char(string="Customer", default='[]')
     has_partner_email = fields.Boolean(compute='_compute_has_partner_email')
+    template_id = fields.Many2one('mail.template', 'Use template', domain="[('model', '=', 'coupon.coupon')]")
```

Comment:
shouldn't we call "_get_default_template"? Because the "if" bellow was probably here in case the xml_id was deleted.

Reply:
Do you mean adding `default=lambda self: self.env['coupon.coupon']._get_default_template()` in the field declaration? The expression will just return `False` since `_get_default_template` is based on the source order of the coupon record. This wizard has no context of source order so the generated coupon won't have any, and as a result, calling `_get_default_template` here will always return `False`.

Reply:
So if I've well understood, here we need a template that is not based on sale_order or pos_order, as it is only to warn partner that they get a coupon. Does such a template exists, or we only have one based on sale_order and on based on pos_order?

Reply:
We only have one for each pos.order and sale.order. User has to choose between the two (if both is installed) in order to send the generated coupons to the customers via email. Now I realized that maybe I can set a default if only one template is available.

Reply:
ok